### DISCUSSION TITLE
[Discussion] Terrain Generation

### DISCUSSION THREAD

**Discussion by ZachZent** [tsunamayo/Starship-EVO#4015] [General]
> Yes I need to redo the tech for ground, using voxels (a different kind though, with LOD). I am still not sure if each section would have its own voxel world or if I would just spawn on single for all the hull sections. In anycase they would all use the same procedural terrain so they would match.

If we could get a bit more information on how you plan to generate terrain, that would be great. There is smooth generation of NMS, Astroneer, or Empyrion on one end and Minecraft full block on the other.

**Comment by tsunamayo** [comment]
Thats is easy, that would be NMS (marching cubes in tech lingo). block would look out of place, and most importantly I need LODs...
More specifically I will have a shot at this implementation:
http://transvoxel.org/

**Reply by Jorpis-Jorah** [reply]
Will Gas giants and/or volumetric clouds ever be implemented?
